lib/framework/debug.cpp

```diff
@@ -34,7 +34,16 @@
 #include "wzapp.h"
 #include <map>
 #include <string>
-#include <regex>
+// On Fedora 40, GCC 14 produces false-positive warnings for -Walloc-zero
+// when compiling <regex> with optimizations. Silence these warnings.
+#if defined(__GNUC__) && __GNUC__ >= 14 && defined(__OPTIMIZE__)
```

Comment:
Because LLVM/Clang and Intel's compiler also define `__GNUC__`, we should follow the pattern used elsewhere and exclude them as follows:

```suggestion
#if !defined(__clang__) && !defined(__INTEL_COMPILER) && defined(__GNUC__) && __GNUC__ >= 14 && defined(__OPTIMIZE__)
```